app/controllers/katello/api/v2/content_view_components_controller.rb

```diff
@@ -0,0 +1,100 @@
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :components, Array, :desc => N_("Array of components to add"), :required => true do
+      param :content_view_version_id, :identifier, :desc => N_("identifier of the version of the component content view")
+      param :content_view_id, :identifier,
+            :desc => N_("content view identifier of the component who's latest version is desired")
+      param :latest, :bool, :desc => N_("true if the latest version of the components content view is desired")
+    end
+    def add_components
+      components = params.require(:components).map do |component|
+        options = {}
+        options[:latest] = ::Foreman::Cast.to_bool(component[:latest]) if component.key?(:latest)
+        options.merge(component.slice(:content_view_version_id, :content_view_id))
+      end
+      @view.add_components(components)
+      @view.save!
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :PUT, "/content_views/:composite_content_view_id/content_view_components/remove",
+                                        N_("Remove components from the content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :component_ids, Array, :desc => N_("Array of component ids to remove"), :required => true
+    def remove_components
+      @view.remove_components(params.require(:component_ids))
+      @view.save!
+      respond_for_index(:collection => index_response, :template => "index")
+    end
+
+    api :GET, "/content_views/:composite_content_view_id/content_view_components/:id", N_("Show a content view component")
+    param :composite_content_view_id, :number, :desc => N_("composite content view numeric identifier"), :required => true
+    param :id, :identifier, :desc => N_("component ID"), :required => true
+    def show
+      respond :resource => @component
+    end
+
+    api :PUT, "/content_views/:composite_content_view_id/content_view_components/:id",
+        N_("Update a component associated with the content view")
+    param :composite_content_view_id, :identifier, :desc => N_("composite content view identifier"), :required => true
+    param :id, :identifier, :desc => N_("component ID"), :required => true
+    param :content_view_version_id, :identifier, :desc => N_("identifier of the version of the component content view")
+    param :latest, :bool, :desc => N_("true if the latest version of the components content view is desired")
+    def update
+      cvv_id = component_params[:content_view_version_id]
+      if component_params.key?(:latest) && component_params.key?(:content_view_version_id)
+        latest = ::Foreman::Cast.to_bool(component_params[:latest])
+        if latest && cvv_id.present?
+          fail HttpErrors::UnprocessableEntity,
```

Comment:
> @parthaa, in addition to @johnpmitsch's point, I'd recommend that we be consistent in wording with the puppet module selection, since it also supports selecting the latest version.
> 
I have updated the templates to match a combination of @johnpmitsch and @bbuckingham 's views in consultation with @ehelms 

Basically the drop down will say "Always Use Latest (Currently 2.0)" when one is updating or adding a content view.  However list will show "Latest(Currently 2.0)" similar to what puppet modules page shows. I also created a redmine issue to update the puppet UI and have it say 'Always Use Latest (Currently 1.4.0)' -> http://projects.theforeman.org/issues/17390